pwsh/buildkimageformats.ps1

```diff
@@ -112,23 +117,33 @@ if ($IsMacOS -and $env:buildArch -eq 'Universal') {
         # Also copy all the vcpkg DLLs on windows, since it's apparently not static by default
         cp "$env:VCPKG_ROOT/installed/$env:VCPKG_DEFAULT_TRIPLET/bin/*.dll" $prefix_out
     } elseif ($IsMacOS) {
-        cp karchive/bin/*.dylib $prefix_out
+        cp karchive/bin/libKF${kfMajorVer}Archive.$kfMajorVer.dylib $prefix_out
     } else {
-        $env:KF5LibLoc = Split-Path -Path (Get-Childitem -Include libKF5Archive.so.5 -Recurse -ErrorAction SilentlyContinue)[0]
-        cp $env:KF5LibLoc/* $prefix_out
+        cp karchive/bin/libKF${kfMajorVer}Archive.so.$kfMajorVer $prefix_out
     }
 }
 
 # Fix linking on macOS
 if ($IsMacOS) {
-    $libDirName = $qtVersion.Major -eq 5 ? 'lib' : '' # empty name results in double slash in path which is intentional
+    $karchLibName = "libKF${kfMajorVer}Archive.$kfMajorVer"
+    $libDirName = $kfMajorVer -le 5 -and $qtVersion.Major -ge 6 ? '' : 'lib' # empty name results in double slash in path which is intentional
 
-    install_name_tool -change "$(Get-Location)/karchive/installed/$libDirName/libKF5Archive.5.dylib" @rpath/libKF5Archive.5.dylib output/kimg_kra.so
-    install_name_tool -change "$(Get-Location)/karchive/installed/$libDirName/libKF5Archive.5.dylib" @rpath/libKF5Archive.5.dylib output/kimg_ora.so
+    install_name_tool -change "$(Get-Location)/karchive/installed/$libDirName/$karchLibName.dylib" "@rpath/$karchLibName.dylib" "$prefix_out/kimg_kra$kimgLibExt"
+    install_name_tool -change "$(Get-Location)/karchive/installed/$libDirName/$karchLibName.dylib" "@rpath/$karchLibName.dylib" "$prefix_out/kimg_ora$kimgLibExt"
 
     if ($IsMacOS -and $env:buildArch -eq 'Universal') {
-        install_name_tool -change "$(Get-Location)/karchive/installed_arm64/$libDirName/libKF5Archive.5.dylib" @rpath/libKF5Archive.5.dylib output/kimg_kra.so
-        install_name_tool -change "$(Get-Location)/karchive/installed_arm64/$libDirName/libKF5Archive.5.dylib" @rpath/libKF5Archive.5.dylib output/kimg_ora.so
+        install_name_tool -change "$(Get-Location)/karchive/installed_arm64/$libDirName/$karchLibName.dylib" "@rpath/$karchLibName.dylib" "$prefix_out/kimg_kra$kimgLibExt"
+        install_name_tool -change "$(Get-Location)/karchive/installed_arm64/$libDirName/$karchLibName.dylib" "@rpath/$karchLibName.dylib" "$prefix_out/kimg_ora$kimgLibExt"
+    }
+}
+
+# Fix linking on Linux
+if ($IsLinux) {
+    patchelf --set-rpath '$ORIGIN' "$prefix_out/libKF${kfMajorVer}Archive.so.$kfMajorVer"
+
+    $files = Get-ChildItem "$prefix_out" -Recurse -Filter "kimg_*$kimgLibExt"
+    foreach ($file in $files) {
+        patchelf --set-rpath '$ORIGIN/../../lib' $file
```

Comment:
This was a pre-existing problem I noticed when testing the Linux appimage; it couldn't load ORA/KRA files due to the rpath not allowing the image plugins to locate KArchive.